### PR TITLE
Store group sigs in the real header, not in a copy

### PR DESCRIPTION
**Pull request by harmony-ek**
(*Block) Header() returns a (shallow) copy of the header, where we were
storing group signatures and signer bitmaps.  This had no effect, and
the on-chain block was still missing these signatures.

### CODE REVIEW

File: core/types/block.go [hunk 1]
```diff
@@ -159,6 +161,28 @@ type Block struct {
 	ReceivedFrom interface{}
 }
 
+// SetPrepareSig sets the block's prepare group signature.
+func (b *Block) SetPrepareSig(sig []byte, signers []byte) {
```
Comment (LeoHChen) [inline core/types/block.go]: please add unit test.

File: core/types/block.go [hunk 2]
```diff
@@ -159,6 +161,28 @@ type Block struct {
...
+		utils.GetLogInstance().Warn("SetPrepareSig: sig size mismatch",
+			"srcLen", len(sig),
+			"dstLen", len(b.header.PrepareSignature))
+	}
+	copy(b.header.PrepareSignature[:], sig[:])
+	b.header.PrepareBitmap = append(signers[:0:0], signers...)
+}
+
+// SetCommitSig sets the block's commit group signature.
+func (b *Block) SetCommitSig(sig []byte, signers []byte) {
```
Comment (LeoHChen) [inline core/types/block.go]: please add unit test.

File: core/types/block.go (hunk 1 above)
Comment (harmony-ek) [reply core/types/block.go]: Added.

File: core/types/block.go (hunk 2 above)
Comment (harmony-ek) [reply core/types/block.go]: Added.

**Comment by harmony-ek** [comment]
@LeoHChen @rlan35 UT added, please let me know if this LGTY.

**Pull request merged**